src/details/details.test.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * @jest-environment ./lib/puppeteer/environment.js
+ */
+/* eslint-env jest */
+
+let browser
+let page
+let baseUrl = 'http://localhost:3000'
+
+beforeAll(async (done) => {
+  browser = global.__BROWSER__
+  page = await browser.newPage()
+  done()
+})
+
+afterAll(async (done) => {
+  await page.close()
+  done()
+})
+
+describe('/components/details', () => {
+  describe('/components/details/preview', () => {
+    it('should add to summary the button role', async () => {
+      await page.goto(baseUrl + '/components/details/preview', { waitUntil: 'load' })
+
+      const summaryRole = await page.evaluate(() => document.body.getElementsByTagName('summary')[0].getAttribute('role'))
+      expect(summaryRole).toBe('button')
+    })
+
+    it('should set the element controlled by the summary using aria-controls', async () => {
+      await page.goto(baseUrl + '/components/details/preview', { waitUntil: 'load' })
+
+      const summaryAriaControls = await page.evaluate(() => document.body.getElementsByTagName('summary')[0].getAttribute('aria-controls'))
+      expect(summaryAriaControls).toBe('details-content-0')
+    })
+
+    it('should set the expanded state of the summary to false using aria-expanded', async () => {
+      await page.goto(baseUrl + '/components/details/preview', { waitUntil: 'load' })
+
+      const summaryAriaExpanded = await page.evaluate(() => document.body.getElementsByTagName('summary')[0].getAttribute('aria-expanded'))
+      expect(summaryAriaExpanded).toBe('false')
+    })
+
+    it('should add a unique id to the hidden content in order to be controlled by the summary', async () => {
```

Comment:
It would be good to have a test for a details element that already has an existing ID, to ensure the existing ID is not overriden.